File: School source codes/Lab 8/PalindromeInt.cs
using System;
using System.Collections.Generic;
using System.Linq;

class MainClass {
  public static int Reverse (int x) {
    int k , reverse = 0, rem;
    while(x!=0) {      
        rem = x%10;        
        reverse = (reverse * 10) + rem;      
        x/=10;}
    k = reverse;
    return k;
  }
  public static int IsPalindrome (int x) {
    int k;
    if (x == Reverse(x))
      {k = 1;}
    else
      {k = 0;}
    return k;
  }
  public static void Main (string[] args) {
    Console.Write("Please enter an integer: ");
    int zed = Convert.ToInt32(Console.ReadLine());
    int zedcheck = IsPalindrome(zed);
    if (zedcheck == 1)
      {Console.WriteLine("That is a palindrome! Good job.");}
    else
      {Console.WriteLine("That is not a palindrome! Try again.");}
  }
}

File: School source codes/Lab 8/PalindromeInt.rb
def Reverse(x)
  reverse = 0
  while(x!=0)    
    rem = x % 10        
    reverse = ((reverse * 10) + rem)      
    x /= 10
  end
  return reverse
end
def IsPalindrome(x)
  if (x == Reverse(x))
    k = 1
  else
    k = 0
  end
  return k
end
print "Please enter an integer: "
zed = gets.chomp.to_i
zedcheck = IsPalindrome(zed)
if (zedcheck == 1)
  puts "That is a palindrome! Good job."
else
  puts "That is not a palindrome! Try again."
end